esp-hal/src/rtc_cntl/mod.rs

```diff
@@ -217,8 +217,12 @@ impl<'d> Rtc<'d> {
         RtcClock::estimate_xtal_frequency()
     }
 
-    /// Read the current value of the rtc time registers.
-    pub fn get_time_raw(&self) -> u64 {
+    /// Read the current raw value of the rtc time registers.
+    ///
+    /// **This function does not take into account the boot time registers, and
```

Comment:
This would need a clearer explanation, after reading this PR I have no idea what these functions are doing

Reply:
Would an explanation of RTC time registers vs boot time registers be helpful? It could be added to the Rtc struct docs.

Reply:
I would suggest some of the functions be renamed to a more informative form, for example:
`get/set_rtc_count_us` instead of `get_rtc_time_us` (this is just a suggestion). Ideally only one pair of functions should be named after `time` so that users can immediately see which one is the right function to use.

Reply:
When naming public API, please read through the API guidelines. There's a whole chapter about [naming](https://rust-lang.github.io/api-guidelines/naming.html) and I'd like to start adhering to those guidelines moving forward.

Reply:
Also while I don't know the team's opinion on this, I think we use fugit's types in places - maybe we could do that here, too, instead of the multiple getters.

Reply:
> I would suggest some of the functions be renamed to a more informative form, for example: `get/set_rtc_count_us` instead of `get_rtc_time_us` (this is just a suggestion). Ideally only one pair of functions should be named after `time` so that users can immediately see which one is the right function to use.

I like the rtc_count name a lot, since it makes it a bit clearer that it will only ever go up. I also plan to add some documentation to the Rtc struct.

> When naming public API, please read through the API guidelines. There's a whole chapter about [naming](https://rust-lang.github.io/api-guidelines/naming.html) and I'd like to start adhering to those guidelines moving forward.

Are you talking about omitting get_ from method names? The API guidelines don't specify how setters should be named, so imo it would be more clear just to keep the get_. Having `time` and `set_time` seems less straightforward than `get_time` and `set_time`

> Also while I don't know the team's opinion on this, I think we use fugit's types in places - maybe we could do that here, too, instead of the multiple getters.

Yeah, only having one getter and setter would be nice.

Reply:
`get_` is one part, yes, but I wouldn't even keep `rtc_` in the name. I'd suggest `fn counter()`, `fn current_time()`, `fn time_since_boot()` - but definitely don't feel pressured to go with these, feel free to wait for a second or third opinion!

Reply:
`current_time` and `time_since_boot` seemed like much better names so I decided to go with them and make `time_since_boot_raw` private. In addition, I made all the public methods use fugit. Does this look good to everyone?